### DISCUSSION TITLE
Net Zero Decentralised Computing

### DISCUSSION THREAD

**Discussion by Shelpin** [akash-network/community#687] [Ideas]
Hi everyone! 

My name is Álex Casas and I am the head of product at [Climatecoin](https://climatecoin.com). We were granted by the Algorand Foundation to jointly develop a mechanism called Green Treasury.

The Green Treasury calculates through an oracle the Co2 emissions of Algorand nodes and **automatically swaps USDC for tokenized high quality carbon credits** which are immediately retired in exchange of an offset certificate. 

Please see a couple of news about it below:

https://cointelegraph.com/news/algorand-aims-to-convert-network-transaction-fees-into-carbon-offsets
<img width="910" alt="Algorand aims to convert network" src="https://github.com/user-attachments/assets/9a0a19ab-0260-4d1f-85c2-ae1f475b3c94">

https://finance.yahoo.com/news/algorand-launch-first-carbon-emission-154644200.html
<img width="1201" alt="image" src="https://github.com/user-attachments/assets/78a10674-6329-46e6-96c3-7e57adfe781f">

AI computing has not got so roasted as Bitcoin for its energy consumption and environmental effect , but that does not mean that that effect is not there, and that imho there is a **powerful value proposition here, Net Zero Decentralized Computing.** 

We would love to help Akash to pioneer sustainable computing, and be the **first mover** of this upcoming trend. **Secure, cost effective and net zero access to enterprise grade GPU's and computing resources.** 

So, I am writing this post as a temperature check about what the Akash community thinks about this possibility, and ideal ways to implement it , all the use of Akash resources being net zero, net zero opt-in for every lease to have its emissions compensation certificate …. an hybrid , phased approach…

Implementing such a mechanism for Akash Network might be a little bit more complex than for Algorand , but also **much more meaningful and disruptive**, making Akash network the first decentralized computing network that has the ability to **automatically offset the emissions caused by the computing workloads**. 

Happy to brainstorm together to accomplish this.

Have a great day everyone!!

**Comment by dominikusbrian** [comment]
Hi @Shelpin thank you for putting this discussion. Yes indeed the energy consumption aspect of compute for blockchain operation as well as AI system are both in the heat of attention these days. Being from a green tech background and been active in Akash Ecosystem for a while, I see the value in having better grasp on the actual carbon footprint of the decentralized supercloud operation like Akash. 

To this end, perhaps there are several things we can address. First is to perform a survey of energy source used by Akash Providers, this could be part of the already existing Akash Auditing process (which currently focus on hardware specs and middleware robustness). Second, having the energy source known, we can then have some sort of conversion factor regarding the CO2 equivalent emission for every kilowatt hours (kWh) of electricity consumed by each provider. Third is then to introduce a "probe" that attempts to read over the state of each provider's energy usage every certain duration (e.g., 10 mins). 

Open Source tools like https://github.com/mlco2/codecarbon and many others will be useful in developing the probe described above. 
After all the stats data are available and been collected for some time, we can then thing of how to reduce them while maintaining performance, Certification of power efficient providers and recognition for their energy savings and reduction of carbon footprint can then be recognized with badge. User that want to support green movement could then opt-in for provider with such badges. 

Awareness about carbon footprint is first need to be established, before doing emission reduction program, and eventually toward Net Zero system, if that's even possible (need to figure out mechanism and incentive for the offset, which is unclear). 

In a typical Carbon Credit marketplace situation, Akash will be currently placed as the demand side for Carbon Credit. I am not quite sure whether ClimateCoin could have some incentive so that Akash can start moving toward offsetting our ecosystem carbon footprint without paying extra carbon tax ? but instead via other swap mechanism with project that are in the supply side ? 

For instance, carbon surplus renewable energy project like wind farms and solar farms requires a lot of compute for simulating their digital twin or control system. Could they become our carbon offset partner ? We can perhaps design a mechanism where Akash Providers that want to offset their carbon footprint to acquire carbon credit from the green energy project through "paying with extra compute".  Notably this is energy efficient because sometimes the providers are not operating in full capacity and would still need to pay the expenditure/utility cost for the idle compute. Utilization of that compute will surely increase the consumption a bit, but the base energy upkeep is actually the big chunk for their cost and carbon footprint. 

Ideally project like EnergyWebX that already have direct attachement to various green energy projects would be ideal candidate for getting the supply side on board. The way I understand it ClimateCoin is more a marketplace and utility system that facilitate the offset process. Looking forward to hearing more from you.

**Comment by 31trainman** [comment]
I like this idea and see potential in it. Have you taken any steps to bring this to the core team members attention and or the Akash community?
